### ISSUE TITLE
ModuleNotFoundError if _ssl module is unavailable

### ISSUE THREAD

**Issue by lordmauve** [python-trio/trio#1740]
The `_ssl` module is optional in Python, ie. building Python succeeds if TLS libraries are unavailable.

Trio attempts to import `ssl` at import time, meaning that trio is unusable if Python is not built with TLS support; importing it fails with the error

```
ModuleNotFoundError: No module named '_ssl'
```

Simply suppressing the ImportError in `trio._ssl` and `trio._highlevel_ssl_helpers` addresses this, allowing it to fail only if you try to use SSL functionality.

**Comment by njsmith** [comment]
Yeah, I've hesitated to try to fix this, since we have no way to test this configuration so any fixes will likely rot again over time, and... in 2020 TLS is not *really* very optional, regardless of what Python's build scripts think :-). How important do you think fixing this would be?

**Comment by lordmauve** [comment]
Not very important. I think the best case that I can make for it is that maybe there are some tiny IoT devices that run Python but don't want to bother with TLS?

Perhaps it could be fixed with an amount of diligence commensurate with how important it is, eg. just a comment like "ssl may not be available, error at time of use instead"?

**Comment by parity3** [comment]
Another perspective: better not to have TLS support (and be explicit about it) than to have something outdated with known exploits? But I guess in that case future versions of python should at least put a stub module in there that just denies all negotiations gracefully
